modules/mcl-disko/default.nix

```
@@ -18,11 +18,15 @@
       options.mcl.disko = {
         enable = mkEnableOption "Enable Module";
 
-        legacyBoot = mkOption {
-          type = types.bool;
-          default = false;
-          example = true;
-          description = "Declare if the configuration is for a Hetzner server or not";
+        primaryPartitionType = mkOption {
+          type = types.enum [
+            "zfs"
+            "legacyBoot"
+            "ext4"
+          ];
```

Perhaps `partitioningPreset` is a better name as the primary partition type for both `zfs` and `zfs-legacy-boot` are zfs.

```suggestion
        partitioningPreset = mkOption {
          type = types.enum [
            "zfs"
            "zfs-legacy-boot"
            "ext4"
          ];
```

---

Resolved